lib/elixir/test/elixir/kernel/typespec_test.exs

```diff
@@ -635,6 +635,145 @@ defmodule Kernel.TypespecTest do
+      (quote do: @type basic_port() :: port()),
+      (quote do: @type basic_reference() :: reference()),
+      (quote do: @type basic_struct() :: struct()),
+      (quote do: @type basic_tuple() :: tuple()),
+      # numbers
+      (quote do: @type basic_float() :: float()),
+      (quote do: @type basic_integer() :: integer()),
+      (quote do: @type basic_neg_integer() :: neg_integer()),
+      (quote do: @type basic_non_neg_integer() :: non_neg_integer()),
+      (quote do: @type basic_pos_integer() :: pos_integer()),
+      # lists
+      (quote do: @type basic_list_type() :: list(integer())),
+      (quote do: @type basic_nonempty_list_type() :: nonempty_list(integer())),
+      (quote do: @type basic_maybe_improper_list_type() :: maybe_improper_list(integer(), atom())),
+      (quote do: @type basic_nonempty_improper_list_type() :: nonempty_improper_list(integer(), atom())),
+      (quote do: @type basic_nonempty_maybe_improper_list_type() :: nonempty_maybe_improper_list(integer(), atom())),
+
+      ## Literals
+      (quote do: @type literal_atom() :: :atom),
+      (quote do: @type literal_integer() :: 1),
+      (quote do: @type literal_integers() :: 1..10),
+      (quote do: @type literal_empty_bitstring() :: <<>>),
+      (quote do: @type literal_size_0() :: (<<_::0>>)),
+      (quote do: @type literal_unit_1() :: (<<_::_ * 1>>)),
+      (quote do: @type literal_size_1_unit_8() :: (<<_::100, _::_ * 256>>)),
+      (quote do: @type literal_function_arity_any() :: (... -> integer())),
+      (quote do: @type literal_function_arity_0() :: (() -> integer())),
+      (quote do: @type literal_function_arity_2() :: (integer(), atom() -> integer())),
+      (quote do: @type literal_list_type() :: [integer()]),
+      (quote do: @type literal_empty_list() :: []),
+      (quote do: @type literal_list_nonempty() :: [...]),
+      (quote do: @type literal_nonempty_list_type() :: [atom(), ...]),
+      (quote do: @type literal_keyword_list_fixed_key() :: [key: integer()]),
+      (quote do: @type literal_keyword_list_fixed_key2() :: [{:key, integer()}]),
+      (quote do: @type literal_keyword_list_type_key() :: [{binary(), integer()}]),
+      (quote do: @type literal_empty_map() :: %{}),
+      (quote do: @type literal_map_with_key() :: %{key: integer()}),
+      (quote do: @type literal_map_with_required_key() :: %{required(bitstring()) => integer()}),
+      (quote do: @type literal_map_with_optional_key() :: %{optional(bitstring()) => integer()}),
+      # TODO: Remove by 1.5
+      (quote do: @type literal_map_with_arrow() :: %{any() => any()}),
+      (quote do: @type literal_struct_all_fields_any_type() :: %SomeStruct{}),
+      (quote do: @type literal_struct_all_fields_key_type() :: %SomeStruct{key: integer()}),
+      (quote do: @type literal_empty_tuple() :: {}),
+      (quote do: @type literal_2_element_tuple() :: {1, 2}),
+
+      ## Built-in types
+      (quote do: @type built_in_term() :: term()),
```

Comment:
We usually don't replace hyphen with underscore, rather simply remove it, e.g. `builtin_term`.